src/orchestrator.rs

```diff
@@ -137,9 +140,20 @@ impl Orchestrator {
                 }
                 Ok(generation_results)
             }
-        })
-        .await
-        .unwrap()
+        });
+        match task_handle.await {
+            Ok(Ok(result)) => Ok(result),
+            Ok(Err(error)) => {
+                error!(request_id = ?task.request_id, %error, "unary task failed");
```

Comment:
I agree, this can be confusing.

Essentially, `task.handle.await` returns `Result<Result<ClassifiedGeneratedTextResult, Error>, JoinError>`. The error type for this outer `Result` is `tokio::task::JoinError`, which is returned if a task is cancelled or panics.

`Ok(Ok(result))` => task completed successfully, so we have a `ClassifiedGeneratedTextResult`
`Ok(Err(error))` => task failed, so we have a `orchestrator::Error` propagated from the child task that failed
`Err(error)` => task was cancelled or panicked, so we have a `tokio::task::JoinError`

I will add this to the comments.